go/vt/vtgate/evalengine/comparisons.go

```diff
@@ -146,7 +146,7 @@ func evalResultsAreDateAndNumeric(l, r *EvalResult) bool {
 func compareAsTuples(lVal, rVal *EvalResult) bool {
 	left := lVal.typeof() == sqltypes.Tuple
 	right := rVal.typeof() == sqltypes.Tuple
-	if left && right {
```

Comment:
Is there a higher chance that the right side will be nil?

Reply:
🤷 

don't know. why do you ask?

Reply:
that was probably just my OCD